scripts/prepare_lima.py

```diff
@@ -0,0 +1,191 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import json
+import sys
+from pathlib import Path
+from typing import Optional
+
+from datasets import load_dataset
+import torch
+from torch.utils.data import random_split
+from tqdm import tqdm
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt.tokenizer import Tokenizer
+
+DATA_REPO_ID = "GAIR/lima"
+DATA_FILE_NAME = "lima_data_cleaned_archive.json"
+DESTINATION_PATH = Path("data/lima")
+CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+TEST_SPLIT_FRACTION = 0.1
+MASK_INPUTS = False  # as in alpaca-lora
+IGNORE_INDEX = -1
+SEED = 42
+
+
+def prepare(
+    destination_path: Path = DESTINATION_PATH,
+    test_split_fraction: float = TEST_SPLIT_FRACTION,
+    checkpoint_dir: Path = CHECKPOINT_DIR,
+    mask_inputs: bool = MASK_INPUTS,
+    seed: int = SEED,
+    include_multiturn_conversations: bool = False,
+    data_repo_id: str = DATA_REPO_ID,
+    ignore_index: int = IGNORE_INDEX,
+    max_seq_length: Optional[int] = None,
+    hf_access_token: str = None
+) -> None:
+    """Prepare the LIMA dataset for instruction tuning.
+
+    The output is a training and test dataset saved as `train.pt` and `test.pt`,
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+
+    if hf_access_token is None:
+        raise ValueError(
+            "This dataset requires authentication. Please provide "
+            "your Hugging Face Access Token via `--hf_access_token`."
+        )
+
+    if max_seq_length is None:
+        with open(checkpoint_dir / "lit_config.json", "r", encoding="utf-8") as file:
+            config = json.load(file)
+            max_seq_length = config["block_size"]
+
+    destination_path.mkdir(parents=True, exist_ok=True)
+    print("Loading data file...")
+
+    dataset = load_dataset(data_repo_id, use_auth_token=hf_access_token)
```

Comment:
The dependency issue will go away if you make the import local

```suggestion
    from datasets import load_dataset
    dataset = load_dataset(data_repo_id, use_auth_token=hf_access_token)
```

Reply:
Ah yes, helps us to reduce install dependencies I guess. Thanks!